views/js/pciCreator/ims/audioRecordingInteraction/runtime/js/uiElements.js

```diff
@@ -336,10 +337,113 @@ define([
+        const soundUrl = beepSoundDataUrl;
+
+        let beepPlayer;
+        let audioEl = new Audio();
+        let playedToTheEndPromise;
+
+        function createDeferredPromise() {
+            const deferred = {};
+            deferred.promise = new Promise((resolve, reject) => {
+                deferred.resolve = resolve;
+                deferred.reject = reject;
+            });
+            return deferred;
+        }
+
+        function playSound() {
+            if (playedToTheEndPromise) {
+                playedToTheEndPromise.reject();
+            }
+            playedToTheEndPromise = createDeferredPromise();
+
+            return Promise.resolve()
+                .then(() => {
+                    if (!audioEl) {
+                        playedToTheEndPromise.reject();
+                        return;
+                    }
+
+                    audioEl.pause();
+                    audioEl.src = soundUrl;
+                    audioEl.currentTime = 0;
+                    audioEl.muted = false;
+
+                    audioEl.onended = () => {
+                        clearTimeout(timeoutId);
+                        if (playedToTheEndPromise) {
+                            playedToTheEndPromise.resolve();
+                        }
+                    };
+                    audioEl.onerror = err => {
+                        clearTimeout(timeoutId);
+                        if (playedToTheEndPromise) {
+                            playedToTheEndPromise.reject(err);
+                        }
+                    };
+                    const timeoutId = setTimeout(() => {
+                        if (playedToTheEndPromise) {
+                            playedToTheEndPromise.reject(new Error('beep was not played in time'));
+                        }
+                    }, timeoutMs);
+
+                    return audioEl.play();
+                })
+                .then(() => playedToTheEndPromise.promise)
+                .catch(function onError(err) {
+                    if (err) {
+                        // eslint-disable-next-line no-console
+                        console.error(err);
+                    }
+                })
+                .finally(() => {
+                    playedToTheEndPromise = null;
+                });
+        }
+
+        function playSilentSound() {
+            if (!playedToTheEndPromise && audioEl) {
+                audioEl.src = soundUrl;
+                audioEl.muted = true;
+                audioEl.play().catch(() => {});
+            }
+        }
+
+        /**
+         * try to workaround the issue where iOS doesn't play audio if play wasn't started by the user action
+         * (happens if recording is set to autostart after the specified delay)
+         */
+        document.body.addEventListener('click', playSilentSound, { once: true });
+
+        beepPlayer = {
+            playStartSound: playSound,
+            playEndSound: playSound,
```

Comment:
It's not so easy, because double beep is long enough to be noticeable. So both dispatch of `recorder-stop` and `autoPlayback` must happen after it finishes playing. 
Working on it.

Reply:
Done